manager/controllers/app/blueprint_controller.go

```diff
@@ -141,6 +155,42 @@ func (r *BlueprintReconciler) applyChartResource(log logr.Logger, chartSpec app.
 	nbytes, _ := yaml.Marshal(args)
 	log.Info(fmt.Sprintf("--- Values.yaml ---\n\n%s\n\n", nbytes))
 
+	if chartSpec.ChartPullSecret != "" {
+		// obtain ChartPullSecret
+		pullSecret := corev1.Secret{}
+		pullSecrets := []corev1.Secret{}
+
+		if err := r.Get(ctx, types.NamespacedName{Namespace: utils.GetSystemNamespace(), Name: chartSpec.ChartPullSecret},
+			&pullSecret); err == nil {
```

Comment:
Do you have RBAC for secrets in system namespace? Is corev1.Secret mapped to the system namespace in manager cache?

Reply:
hi @shlomitk1 : thank you for helping me confirm that we have RBAC for secrets in system namespace